Tests.Unit.Core.ApplicationServices/ApplicationServices/Contract/ItContractWriteServiceTest.cs

```diff
@@ -1030,6 +1027,191 @@ private static bool MatchExpectedRoleAssignments(IEnumerable<(Guid, Guid)> input
+                PaymentFrequencyUuid = A<Guid?>().AsChangedValue()
+            };
+            var (organizationUuid, parameters, createdContract, transaction) = SetupCreateScenarioPrerequisites(paymentModel: paymentModel);
+            var operationError = A<OperationError>();
+            ExpectUpdateIndependentOptionTypeAssignmentReturns<PaymentFreqencyType>(createdContract, paymentModel.PaymentFrequencyUuid.NewValue, operationError);
+            
+            //Act
+            var result = _sut.Create(organizationUuid, parameters);
+
+            //Assert
+            AssertFailureWithKnownError(result, operationError, transaction);
+        }
+
+        [Fact]
+        public void Cannot_Create_With_PaymentModel_If_UpdatePaymentModelType_Fails()
+        {
+            //Arrange
+            var paymentModel = new ItContractPaymentModelModificationParameters()
+            {
+                PaymentModelUuid = A<Guid?>().AsChangedValue()
+            };
+            var (organizationUuid, parameters, createdContract, transaction) = SetupCreateScenarioPrerequisites(paymentModel: paymentModel);
+            var operationError = A<OperationError>();
+            ExpectUpdateIndependentOptionTypeAssignmentReturns<PaymentModelType>(createdContract, paymentModel.PaymentModelUuid.NewValue, operationError);
+
+            //Act
+            var result = _sut.Create(organizationUuid, parameters);
+
+            //Assert
+            AssertFailureWithKnownError(result, operationError, transaction);
+        }
+
+        [Fact]
+        public void Cannot_Create_With_PaymentModel_If_UpdatePriceRegulationType_Fails()
+        {
+            //Arrange
+            var paymentModel = new ItContractPaymentModelModificationParameters()
+            {
+                PriceRegulationUuid = A<Guid?>().AsChangedValue()
+            };
+            var (organizationUuid, parameters, createdContract, transaction) = SetupCreateScenarioPrerequisites(paymentModel: paymentModel);
+            var operationError = A<OperationError>();
+            ExpectUpdateIndependentOptionTypeAssignmentReturns<PriceRegulationType>(createdContract, paymentModel.PriceRegulationUuid.NewValue, operationError);
+
+            //Act
+            var result = _sut.Create(organizationUuid, parameters);
+
+            //Assert
+            AssertFailureWithKnownError(result, operationError, transaction);
+        }
+
+        [Fact]
+        public void Cannot_Create_With_PaymentModel_If_PaymentMilestones_Does_Not_Have_Either_Date_Set()
+        {
+            //Arrange
+            var milestone = CreatePaymentMilestone(false, false);
+            var paymentModel = new ItContractPaymentModelModificationParameters()
+            {
+                PaymentMileStones = new List<ItContractPaymentMilestone>{ milestone }
+            };
+            var (organizationUuid, parameters, createdContract, transaction) = SetupCreateScenarioPrerequisites(paymentModel: paymentModel);
+
```

Comment:
vi får lidt dobbelttest af den der "AddPaymentMilestone" funktion. Vi kan overveje om vi virkelig skal det, for en eventuel ændring kræver jo så at vi dækker det begge steder. Havde vores repositories afleveret interfaces i stedet for konkrete implementationer ville det være en anden sag, men her har vi i praksis dobbelt test og vi er nødt til det i application service for vi skal se at den kortslutter flowet ved fejl. Spørgsmålet er så om vi behøves de samme tests i itcontracttests.cs